bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5201,7 +5184,31 @@ void ReplicaImp::finishExecutePrePrepareMsg(PrePrepareMsg *ppMsg,
 
   tryToStartOrFinishExecution(false);
 }
-
+void ReplicaImp::handleDbCheckpointCreation(const SeqNum sn, const uint64_t timestamp) {
+  const auto consensus_time_ms = ConsensusTime(timestamp);
+  auto timeSinceLastSnapshot = (std::chrono::duration_cast<std::chrono::seconds>(consensus_time_ms) -
+                                DbCheckpointManager::instance().getLastCheckpointCreationTime())
+                                   .count();
+  const auto &seq_num_to_create_dbcheckpoint = DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot();
+  if ((timeSinceLastSnapshot >= getReplicaConfig().dbSnapshotIntervalSeconds.count() &&
+       sn - DbCheckpointManager::instance().getLastDbCheckpointSeqNum() >= getReplicaConfig().dbCheckPointWindowSize) ||
+      (seq_num_to_create_dbcheckpoint.has_value() && *seq_num_to_create_dbcheckpoint == sn)) {
+    std::optional blockId(DbCheckpointManager::instance().getLastReachableBlock());
+    LOG_INFO(GL,
+             "creating db checkpoint in the next stable sequence number" << KVLOG(timeSinceLastSnapshot, sn, *blockId));
+    DbCheckpointManager::instance().saveDbMetadataToReservedPages(
+        sn, std::chrono::duration_cast<std::chrono::seconds>(consensus_time_ms));
+    // The above should be happen determenistically among all replicas, regardless if they actually create the
```

Comment:
Since readers usually read from top to bottom, can you move the comment so that it precedes the line it comments?